libqtile/utils.py

```diff
@@ -209,3 +209,38 @@ def send_notification(title, message, urgent=False, timeout=10000):
         notifier.show()
     except Exception as exception:
         logger.error(exception)
+
+
+def guess_terminal():
+    """Try to guess terminal."""
+    test_terminals = [
+            ['x-terminal-emulator', '-?'],
+            ['roxterm', '-?'],
+            ['sakura', '-v'],
+            ['hyper', '-v'],
+            ['alacritty', '-V'],
+            ['terminator', '-v'],
+            ['gnome-terminal', '-?'],
+            ['konsole', '--help'],
+            ['xfce4-terminal', '-V'],
+            ['lxterminal', '-v'],
+            ['mate-terminal', '-h'],
+            ['tilda', '--help'],
+            ['guake', '-h'],
+            ['eterm', '-h'],
+            ['st', '-v'],
+            ['urxvt', '-help'],
+            ['xterm', '-help'],
+            ]
+
+    to_null = ['>', '/dev/null']
+    for terminal in test_terminals:
+        logger.debug('Guessing terminal: {}'.format(terminal[0]))
+        exit_code = os.spawnvp(os.P_WAIT, terminal[0], terminal + to_null)
```

Comment:
Might be worth using subprocess here so we don't need the shell intermediary to do the redirection. But also maybe it doesn't matter.

Reply:
I was expecting this comment :)

Reply:
I thought about subprocess, but tried this way